himan-scripts/CB-TCU-cloud.lua

```diff
@@ -59,11 +80,64 @@ if not NL or not NM or not RR then
   return
 end
 
+if currentProducerName == "MEPS" or currentProducerName == "MEPSMTA" then
+  local Nmat = matrixf(result:GetGrid():GetNi(), result:GetGrid():GetNj(), 1, 0)
+  Nmat:SetValues(EL500)
+  EL500 = Max2D(Nmat,filter,configuration:GetUseCuda()):GetValues()
+
+  Nmat:SetValues(pEL500)
+  pEL500 = Min2D(Nmat,filter,configuration:GetUseCuda()):GetValues()
```

Comment:
Why take smallest pressure (highest level measured from ground up) instead of say average?

Reply:
The macro is using the Max2D EL500 height above ground in the filter area. So then for the EL500 pressure I use Min2D as pressure decreases with altitude. Instead of taking the Max2D of EL500 and converting the height to pressure.

Reply:
If you mean why the macro is using the Max instead of Avg I assume it is done so that airplanes stay above convection at any time. You know the tale of the horse that drowned in a river that was on average 1m deep.